### PR TITLE
Migrate NUnit assertions to the constraint model in preparation of the NUnit 4 upgrade

### PR DESCRIPTION
**Pull request by cquirosj**
This PR contains all changes related to assertion model according to the [constraint assertion model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertion-models/constraint.html).

Using a script, we apply all [assertion rules](https://docs.nunit.org/articles/nunit-analyzers/NUnit-Analyzers.html#assertion-rules-nunit2001---) provided by the NUnit.Analyzers package. In addition, some manual interventions are done when applicable.

## Plan of action

- [x] ⚠️ Build the solution and resolve any remaining errors or warnings
- [x] ✅ Review the assertions commit by commit
- [x] 👀 Invite another pair of eyes to re-review
- [x] 🏗 Set the PR to _In progress_ on the board

### CODE REVIEW

File: src/NServiceBus.Gateway.AcceptanceTests/NServiceBus.Gateway.AcceptanceTests.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.11.0" PrivateAssets="All" />
     <PackageReference Include="NUnit3TestAdapter" Version="4.5.0" PrivateAssets="All" />
     <PackageReference Include="Particular.Packaging" Version="4.1.0" PrivateAssets="All" />
+    <PackageReference Include="NUnit.Analyzers" Version="4.3.0" />
```
Comment (danielmarbach) [inline src/NServiceBus.Gateway.AcceptanceTests/NServiceBus.Gateway.AcceptanceTests.csproj:21]:
```suggestion
    <PackageReference Include="NUnit.Analyzers" Version="4.3.0" PrivateAssets="All" />
```

File: src/NServiceBus.Gateway.PersistenceTests/NServiceBus.Gateway.PersistenceTests.csproj [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
   <ItemGroup>
     <PackageReference Include="NUnit" Version="[3.13.3, 4.0.0)" />
+    <PackageReference Include="NUnit.Analyzers" Version="4.3.0" />
```
Comment (danielmarbach) [inline src/NServiceBus.Gateway.PersistenceTests/NServiceBus.Gateway.PersistenceTests.csproj:15]: We need to bump NUnit and move the analyzer to private assets

**Review by danielmarbach — CHANGES_REQUESTED** [review]

File: src/NServiceBus.Gateway.PersistenceTests/NServiceBus.Gateway.PersistenceTests.csproj (hunk 2 above)
Comment (danielmarbach) [reply src/NServiceBus.Gateway.PersistenceTests/NServiceBus.Gateway.PersistenceTests.csproj:15]: Happy to do that once I find some time today

**Review by lailabougria — APPROVED** [review]
Approving as I see the private assets are set already

**Review by danielmarbach — APPROVED** [review]

**Pull request merged**